src/core/raster/qgssinglebandpseudocolorrenderer.cpp

```diff
@@ -84,6 +88,193 @@ void QgsSingleBandPseudoColorRenderer::setShader( QgsRasterShader* shader )
   mShader = shader;
 }
 
+void QgsSingleBandPseudoColorRenderer::createShader( QgsColorRamp* colorRamp, QgsColorRampShader::ColorRamp_TYPE colorRampType, bool clip )
```

Comment:
@wonder-sk , ok, I'll work something out based on this. The min/max might have to stay for other purposes.